platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1363,10 +1359,13 @@ public void onSurfaceCreated(final GL10 gl, final EGLConfig config) {
     // ==================
     @Keep
     void cancelUrlRequest(final long requestHandle) {
-        if (httpHandler == null) {
-            return;
+        if (httpHandler != null) {
+            Object request = httpRequestHandles.get(requestHandle);
+            if (request != null) {
+                httpHandler.cancelRequest(request);
+            }
+            httpRequestHandles.remove(requestHandle);
```

Comment:
Requests and responses will most likely be occurring on different threads and I don't think `LongSparseArray` is thread-safe, so we should synchronize access to it.